implementation.js

```diff
@@ -149,28 +161,14 @@ module.exports = function from(arrayLike) {
 		}
 	}
 
-	var len = ES.ToLength(items.length);
+	var arrayLike = ES.ToObject(parseIterableLike(items));
+	var len = ES.ToLength(arrayLike.length);
 	var A = ES.IsCallable(C) ? ES.ToObject(new C(len)) : new Array(len);
```

Comment:
Sure, if you're willing to make that PR, then I think this can be merged in the meantime.